tests/conftest.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+from importlib.metadata import version
+from unittest.mock import PropertyMock
+
+import pytest
+from ops import JujuVersion
+from pytest_mock import MockerFixture
+
+
+@pytest.fixture(autouse=True)
+def juju_has_secrets(mocker: MockerFixture):
+    """This fixture will force the usage of secrets whenever run on Juju 3.x.
+
+        NOTE: This is needed, as normally JujuVersion is set to 0.0.0 in tests
+    (i.e. not the real juju version)
+    """
+    if version("juju") < "3":
+        mocker.patch.object(
+            JujuVersion, "has_secrets", new_callable=PropertyMock
+        ).return_value = False
+        return False
+    else:
+        mocker.patch.object(
+            JujuVersion, "has_secrets", new_callable=PropertyMock
+        ).return_value = True
+        return True
```

Comment:
FYI, the above function [got enhanced in Opensearch](https://github.com/canonical/opensearch-operator/blob/main/tests/conftest.py#L13) to look at the `LIBJUJU_VERSION_SPECIFIER`

Thus no libjuju installation is needed for unittests to use the fixture

Reply:
Because we are not supporting Juju 2.9 this will be removed in the separate PR